File: csharp/collatz-conjecture/CollatzConjecture.cs
﻿using System;

public static class CollatzConjecture
{
	public static int Steps(int number)
	{
		if (number < 1)
		{
			throw new ArgumentException();
		}

		int steps = 1;

		while (number != 1)
		{
			if (number % 2 == 0)
				number /= 2;
			else
				number = number * 3 + 1;

			steps++;
		}

		return steps; 
	}
}

File: ruby/collatz-conjecture/collatz_conjecture.rb
class CollatzConjecture
  def self.steps(n)
    raise ArgumentError if n < 1
    return 0 if n == 1
    steps = 0
    until n == 1
      steps += 1
      n = n.even? ? n / 2 : n * 3 + 1;
    end
    steps
  end
end

module BookKeeping
  VERSION = 1
end
